docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,438 @@
+- Download a popular model from HF Hub
+- Fine-tune the model using a relevant fine-tuning technique. The exact technique used
+  will depend on factors such as the model, amount and nature of training data, your hardware
+  setup and the end task for which the model will be used
+- Evaluate the model on some benchmarks to validate model quality
+- Run some generations to make sure the model output looks reasonable
+- Quantize the model for efficient inference followed by optionally exporting it for specific
+  environments such as inference on a mobile phone
+
+In this tutorial we'll cover each of these items and give examples of how you can do this using
+TorchTune, and how TorchTune makes it easy to use popular tools and libraries from the ecosystem.
+
+We'll use the Llama2 7B model for this tutorial. You can find a complete set of models supported
+by TorchTune `here <https://github.com/pytorch/torchtune/blob/main/README.md#introduction>`_.
+
+|
+
+Download Llama2 7B
+------------------
+
+In this tutorial we'll use the HF-Format for the Llama2 7B model. For more information on checkpoint
+formats and how these are handled in TorchTune, take a look at this tutorial on checkpointing.
+
+To download the HF format Llama2 7B model, we'll use the tune CLI.
+
+.. code-block:: bash
+
+  tune download \
+  meta-llama/Llama-2-7b \
+  --output-dir <checkpoint_dir> \
+  --hf-token <ACCESS TOKEN>
+
+Make a note of <checkpoint_dir>, we'll use this many times in this tutorial.
+
+|
+
+Finetune the model using LoRA
+-----------------------------
+
+For this tutorial, we'll fine-tune the model using LoRA. LoRA is a parameter efficient fine-tuning
+technique which is especially helpful when you don't have a lot of GPU memory to play with. LoRA
+freezes the base LLM and adds a very small percentage of learnable parameters. This helps keep
+memory associated with gradients and optimizer state low. Using TorchTune, you should be able to
+fine-tune a Llama2 7B model with LoRA in less than 16GB of GPU memory using bfloat16 on a
+RTX 3090/4090. For more information on how to use LoRA, take a look at our
+:ref:`LoRA Tutorial <lora_finetune_label>`.
+
+We'll fine-tune using our
+`single device LoRA recipe <https://github.com/pytorch/torchtune/blob/main/recipes/lora_finetune_single_device.py>`_
+and use the standard settings from the
+`default config <https://github.com/pytorch/torchtune/blob/main/recipes/configs/llama2/7B_lora_single_device.yaml>`_.
+
+This will fine-tune our model on the
+`Alpaca Dataset <https://github.com/pytorch/torchtune/blob/main/torchtune/datasets/_alpaca.py>`_
+using a ``batch_size`` of ``2`` and ``dtype`` of ``bfloat16``. With these settings the model
+should have a peak memory usage of ~16GB and total training time of around 2hours for each epoch.
+We'll need to make some changes to the config to make sure our recipe can access the
+right checkpoints.
+
+Let's first copy over the config to our local working director so we can make changes.
```

Comment:
```suggestion
Let's first copy over the config to our local working directory so we can make changes.
```

Reply:
Also somewhere before this I wonder if you wanna sneak in a usage of `tune ls` for completeness. This could be nice exposition on how someone would even know that `llama2/7B_lora_single_device` is the config they want

Reply:
Great point!